### PR TITLE
feat(trace-view): Add an assistive guide to trace view

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds an assistive guide to trace view to show what it can do.

### CODE REVIEW

File: static/app/views/performance/traceDetails/content.tsx [hunk 1]
```diff
@@ -357,13 +360,15 @@ class TraceDetailsContent extends React.Component<Props, State> {
       index,
       numberOfHiddenTransactionsAbove,
       traceInfo,
+      hasGuideAnchor = true,
```
Comment (k-fish) [inline static/app/views/performance/traceDetails/content.tsx:363]: Minor nit.. With the default I think you are better either removing the undefined type on the `hasGuideAnchor` prop or switching the prop to be `isGuideAnchorDisabled` so it doesn't need a default and you can leave it undefined minimizing the function param surface area. Just an aside though, this _might_ simplify the logic in a few places (like with disabled props to the guideanchor below, etc.)

File: src/sentry/assistant/guides.py [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "release_adoption": {"id": 13, "required_targets": ["release_adoption"]},
     "user_misery": {"id": 14, "required_targets": ["user_misery"]},
     "stack_trace_preview": {"id": 15, "required_targets": ["issue_stream_title"]},
+    "trace_view": {"id": 16, "required_targets": []},
```
Comment (k-fish) [inline src/sentry/assistant/guides.py:31]: Why are required targets required below as `['trace_view_guide_row', 'trace_view_guide_row_details']` but none here?

**Review by k-fish — APPROVED** [review]
I don't have enough context on guides to be 100%, but code-wise looks fine.

File: src/sentry/assistant/guides.py (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/assistant/guides.py]: Good catch, fixed in be26d44

File: static/app/views/performance/traceDetails/content.tsx (hunk 1 above)
Comment (Zylphrex) [reply static/app/views/performance/traceDetails/content.tsx]: i removed the `undefined` type on this and made it required.

**Pull request merged**